### PR TITLE
refactor: cleaning up API tests

### PR DESCRIPTION
**Pull request by vilkg**
This PR was born while writing API guide. I went through all the tests and found some areas to improve: 
1. Code formatting, styling issues
2. Configuration handling. Configuration will be taken from config.properties if there's no matching system properties. Hopefully that will be more straight-forward and people will know where to look for it. 
3. Extracted some convenience methods. 
4. Added additional assertions to improve reporting 

Will try to back-port this as well, especially config management.

### CODE REVIEW

File: dhis-2/dhis-e2e-test/src/main/java/org/hisp/dhis/helpers/QueryParamsBuilder.java [hunk 1]
```diff
@@ -40,48 +40,55 @@
...
 
         return this;
     }
 
-    public QueryParamsBuilder addAll(String... params) {
-        for ( String param: params
-               )
+    public QueryParamsBuilder addAll( String... params )
+    {
+        for ( String param : params
```
Comment (stian-sandvold) [inline dhis-2/dhis-e2e-test/src/main/java/org/hisp/dhis/helpers/QueryParamsBuilder.java]:
```suggestion
        for ( String param : params )
```

File: dhis-2/dhis-e2e-test/src/main/java/org/hisp/dhis/helpers/QueryParamsBuilder.java [hunk 2]
```diff
@@ -40,48 +40,55 @@
...
+        )
```
Comment (stian-sandvold) [inline dhis-2/dhis-e2e-test/src/main/java/org/hisp/dhis/helpers/QueryParamsBuilder.java]:
```suggestion

```

File: dhis-2/dhis-e2e-test/src/main/java/org/hisp/dhis/helpers/config/Config.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+package org.hisp.dhis.helpers.config;
+
+/**
+ * @author Gintare Vilkelyte <vilkelyte.gintare@gmail.com>
+ */
+@org.aeonbits.owner.Config.LoadPolicy( org.aeonbits.owner.Config.LoadType.MERGE )
+@Config.Sources( { "system:properties", "system:env", "classpath:config.properties" } )
+public interface Config
+
```
Comment (stian-sandvold) [inline dhis-2/dhis-e2e-test/src/main/java/org/hisp/dhis/helpers/config/Config.java]:
```suggestion
```

File: dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/helpers/extensions/MetadataSetupExtension.java [hunk 4]
```diff
@@ -64,43 +64,54 @@ public void beforeAll( ExtensionContext context )
...
+                "src/test/resources/setup/metadata.json",
+                "src/test/resources/setup/metadata.json",
+                "src/test/resources/setup/users.json"
+            };
 
-            metadataActions.importAndValidateMetadata( new File( "src/test/resources/setup/metadata.json" ), "async=false" );
-            metadataActions.importAndValidateMetadata( new File( "src/test/resources/setup/metadata.json" ), "async=false" );
+            String queryParams = "async=false";
+            for ( String fileName : files
+            )
```
Comment (stian-sandvold) [inline dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/helpers/extensions/MetadataSetupExtension.java]:
```suggestion
```

File: dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/helpers/extensions/MetadataSetupExtension.java [hunk 5]
```diff
@@ -64,43 +64,54 @@ public void beforeAll( ExtensionContext context )
...
+                "src/test/resources/setup/userGroups.json",
+                "src/test/resources/setup/metadata.json",
+                "src/test/resources/setup/metadata.json",
+                "src/test/resources/setup/users.json"
+            };
 
-            metadataActions.importAndValidateMetadata( new File( "src/test/resources/setup/metadata.json" ), "async=false" );
-            metadataActions.importAndValidateMetadata( new File( "src/test/resources/setup/metadata.json" ), "async=false" );
+            String queryParams = "async=false";
+            for ( String fileName : files
```
Comment (stian-sandvold) [inline dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/helpers/extensions/MetadataSetupExtension.java]:
```suggestion
            for ( String fileName : files )
```

**Comment by vilkg** [comment]
Formatting fixed.

**Pull request merged**